test/celeritas/field/DormandPrinceStepper.test.cu

```diff
@@ -0,0 +1,318 @@
+
+namespace celeritas
+{
+namespace test
+{
+//---------------------------------------------------------------------------//
+
+namespace
+{
+//---------------------------------------------------------------------------//
+// KERNELS
+//---------------------------------------------------------------------------//
+template<class Stepper_impl>
+__device__ FieldStepperResult run_stepper(Stepper_impl stepper,
+                                          int step,
+                                          OdeState state,
+                                          int number_threads,
+                                          OdeState* ks,
+                                          OdeState* along_state,
+                                          FieldStepperResult* result)
+{
+    return FieldStepperResult();
+}
+
+template<>
+__device__ FieldStepperResult run_stepper(Stepper_uni stepper,
+                                          int step,
+                                          OdeState state,
+                                          int number_threads,
+                                          OdeState* ks,
+                                          OdeState* along_state,
+                                          FieldStepperResult* result)
+{
+    return stepper(step, state);
+}
+
+template<>
+__device__ FieldStepperResult run_stepper(Stepper_multi_global stepper,
+                                          int step,
+                                          OdeState state,
+                                          int number_threads,
+                                          OdeState* ks,
+                                          OdeState* along_state,
+                                          FieldStepperResult* result)
+{
+    return stepper(step, state, number_threads, ks, along_state, result);
+}
+
+template<>
+__device__ FieldStepperResult run_stepper(Stepper_multi_shared stepper,
+                                          int step,
+                                          OdeState state,
+                                          int number_threads,
+                                          OdeState* ks,
+                                          OdeState* along_state,
+                                          FieldStepperResult* result)
+{
+    return stepper(step, state, number_threads);
+}
+
+template<class Stepper_impl>
+__global__ void dormand_test_arg_kernel(OdeState* states,
+                                        FieldStepperResult* results,
+                                        int* number_iterations,
+                                        int* number_threads,
+                                        OdeState* ks,
+                                        OdeState* along_state)
+{
+    constexpr double initial_step_size = 10000.0;
+    constexpr double delta_chord = 1e-4;
+    constexpr double half = 0.5;
+
+    auto id = (blockIdx.x * blockDim.x + threadIdx.x) / *number_threads;
```

Comment:
you can use `KernelParamCalculator::thread_id()` instead of using builtin CUDA variables.